### ISSUE TITLE
JS routing error when displaying media

### ISSUE THREAD

**Issue by eekes** [sulu/sulu#5283]
| Q | A
| --- | ---
| Sulu Version | ^2.1@dev
| Browser Version | Browser name and version

# Actual Behavior

Since updating to the latest 2.1@dev version every Media view throws these JS errors: 
```
[mobx] Encountered an uncaught exception that was thrown by a reaction or observer component, in: 'Reaction[Autorun@170]' Error: "The route "sulu_media.cget_media does not exist." 
```

Folders are shown, media files aren't.

Tried clearing cache, rebuilding the assets etc. but didn't seem to fix it.

# Expected Behavior

Display the files

# Steps to Reproduce

Another user on Slack stated that just doing a fresh install gave him the same error.

**Comment by mareksuscak** [comment]
Same issue here, except I've just created a new project based on the latest `sulu/skeleton`.

**Comment by alexander-schranz** [comment]
@mareksuscak @eekes can one of you share us the output of `composer info`

**Comment by mareksuscak** [comment]
```
aferrandini/urlizer                  1.0.0      Library for urlizer
antishov/doctrine-extensions-bundle  v1.4.0     Forked from stof/doctrine-extensions-bundle integration of the gedmo/doctrine-extensions with Symfony2
behat/transliterator                 v1.3.0     String transliterator
clue/stream-filter                   v1.4.1     A simple and modern approach to stream filtering in PHP
contao/imagine-svg                   0.2.2      Contao Imagine SVG library
dantleech/glob-finder                1.0.0      Library offering object location from hierarchrical persistent storage systems using globs
dantleech/phpcr-migrations-bundle    1.2.0      PHPCR migrations bundle
doctrine/annotations                 1.10.2     Docblock Annotations Parser
doctrine/cache                       1.10.0     PHP Doctrine Cache library is a popular cache implementation that supports many different drivers such as redis, memcache, apc, mongodb and others.
doctrine/collections                 1.6.4      PHP Doctrine Collections library that adds additional functionality on top of PHP arrays.
doctrine/common                      2.12.0     PHP Doctrine Common project is a library that provides additional functionality that other Doctrine projects depend on such as better reflection support,...
doctrine/data-fixtures               1.4.2      Data Fixtures for all Doctrine Object Managers
doctrine/dbal                        2.10.2     Powerful PHP database abstraction layer (DBAL) with many features for database schema introspection and management.
doctrine/doctrine-bundle             1.12.8     Symfony DoctrineBundle
doctrine/doctrine-cache-bundle       1.4.0      Symfony Bundle for Doctrine Cache
doctrine/doctrine-fixtures-bundle    3.3.1      Symfony DoctrineFixturesBundle
doctrine/event-manager               1.1.0      The Doctrine Event Manager is a simple PHP event system that was built to be used with the various Doctrine projects.
doctrine/inflector                   1.4.0      PHP Doctrine Inflector is a small library that can perform string manipulations with regard to upper/lowercase and singular/plural forms of words.
doctrine/instantiator                1.3.0      A small, lightweight utility to instantiate objects in PHP without invoking their constructors
doctrine/lexer                       1.2.0      PHP Doctrine Lexer parser library that can be used in Top-Down, Recursive Descent Parsers.
doctrine/orm                         v2.7.2     Object-Relational-Mapper for PHP
doctrine/persistence                 1.3.7      The Doctrine Persistence project is a set of shared interfaces and functionality that the different Doctrine object mappers share.
doctrine/phpcr-bundle                2.1.1      Symfony DoctrinePHPCRBundle
doctrine/reflection                  1.2.1      The Doctrine Reflection project is a simple library used by the various Doctrine projects which adds some additional functionality on top of the reflecti...
dragonmantank/cron-expression        v2.3.0     CRON for PHP: Calculate the next or previous run date and determine if a CRON expression is due
egulias/email-validator              2.1.17     A library for validating emails against several RFCs
elasticsearch/elasticsearch          v6.7.2     PHP Client for Elasticsearch
firebase/php-jwt                     v5.2.0     A simple library to encode and decode JSON Web Tokens (JWT) in PHP. Should conform to the current spec.
friendsofsymfony/http-cache          2.9.1      Tools to manage HTTP caching proxies with PHP
friendsofsymfony/http-cache-bundle   2.8.0      Set path based HTTP cache headers and send invalidation requests to your HTTP cache
friendsofsymfony/jsrouting-bundle    2.5.4      A pretty nice way to expose your Symfony2 routing to client applications.
friendsofsymfony/rest-bundle         2.7.4      This Bundle provides various tools to rapidly develop RESTful API's with Symfony
gedmo/doctrine-extensions            v2.4.40    Doctrine2 behavioral extensions
goodby/csv                           1.3.0      CSV import/export library
google/auth                          v1.8.0     Google Auth Library for PHP
google/cloud-core                    v1.36.2    Google Cloud PHP shared dependency, providing functionality useful to all components.
google/cloud-storage                 v1.20.1    Cloud Storage Client for PHP
google/crc32                         v0.1.0     Various CRC32 implementations
guzzlehttp/guzzle                    6.5.3      Guzzle is a PHP HTTP client library
guzzlehttp/promises                  v1.3.1     Guzzle promises library
guzzlehttp/psr7                      1.6.1      PSR-7 message implementation that also provides common utility methods
guzzlehttp/ringphp                   1.1.1      Provides a simple API and specification that abstracts away the details of HTTP into a single PHP function.
guzzlehttp/streams                   3.0.0      Provides a simple abstraction over streams of data
handcraftedinthealps/zendsearch      2.0.0      a general purpose text search engine written entirely in PHP 5
hoa/compiler                         3.17.08.08 The Hoa\Compiler library.
hoa/consistency                      1.17.05.02 The Hoa\Consistency library.
hoa/event                            1.17.01.13 The Hoa\Event library.
hoa/exception                        1.17.01.16 The Hoa\Exception library.
hoa/file                             1.17.07.11 The Hoa\File library.
hoa/iterator                         2.17.01.10 The Hoa\Iterator library.
hoa/math                             1.17.05.16 The Hoa\Math library.
hoa/protocol                         1.17.01.14 The Hoa\Protocol library.
hoa/regex                            1.17.01.13 The Hoa\Regex library.
hoa/stream                           1.17.02.21 The Hoa\Stream library.
hoa/ustring                          4.17.01.16 The Hoa\Ustring library.
hoa/visitor                          2.17.01.16 The Hoa\Visitor library.
hoa/zformat                          1.17.01.10 The Hoa\Zformat library.
http-interop/http-factory-guzzle     1.0.0      An HTTP Factory using Guzzle PSR7
imagine/imagine                      1.2.3      Image processing for PHP 5.3
jackalope/jackalope                  1.4.0      Jackalope PHPCR library
jackalope/jackalope-doctrine-dbal    1.4.1      Jackalope Transport library for Doctrine DBAL
jdorn/sql-formatter                  v1.2.17    a PHP SQL highlighting library
jean85/pretty-package-versions       1.2        A wrapper for ocramius/package-versions to get pretty versions strings
jms/metadata                         2.1.0      Class/method/property metadata management in PHP
jms/serializer                       3.6.0      Library for (de-)serializing data of any complexity; supports XML, JSON, and YAML.
jms/serializer-bundle                3.5.0      Allows you to easily serialize, and deserialize data of any complexity
layershifter/tld-database            1.0.69     Database abstraction for Public Suffix List
layershifter/tld-extract             2.0.1      TLDExtract, library for extracting parts of domain, e.q. domain parser
layershifter/tld-support             1.1.1      Support package for TLDDatabase and TLDExtract
league/flysystem                     1.0.67     Filesystem abstraction: Many filesystems, one API.
massive/build-bundle                 0.4.0      Massive Art Build Bundle
massive/search-bundle                2.2.2      Massive Search Bundle
monolog/monolog                      1.25.3     Sends your logs to files, sockets, inboxes, databases and various web services
mtdowling/cron-expression            v1.2.3     CRON for PHP: Calculate the next or previous run date and determine if a CRON expression is due
mustangostang/spyc                   0.6.3      A simple YAML loader/dumper class for PHP
ocramius/package-versions            1.5.1      Composer plugin that provides efficient querying for installed package versions (no runtime IO)
ocramius/proxy-manager               2.2.3      A library providing utilities to generate, instantiate and generally operate with Object Proxies
ongr/elasticsearch-bundle            v5.2.6     Elasticsearch bundle for Symfony.
ongr/elasticsearch-dsl               v6.1.2     Elasticsearch DSL library
oro/doctrine-extensions              1.3.0      Doctrine Extensions for MySQL and PostgreSQL.
pagerfanta/pagerfanta                v2.1.3     Pagination for PHP
php-http/client-common               2.1.0      Common HTTP Client implementations and tools for HTTPlug
php-http/discovery                   1.7.4      Finds installed HTTPlug implementations and PSR-7 message factories
php-http/guzzle6-adapter             v2.0.1     Guzzle 6 HTTP Adapter
php-http/httplug                     2.1.0      HTTPlug, the HTTP client abstraction for PHP
php-http/message                     1.8.0      HTTP Message related tools
php-http/message-factory             v1.0.2     Factory interfaces for PSR-7 HTTP Message
php-http/promise                     v1.0.0     Promise used for asynchronous HTTP requests
phpcr/phpcr                          2.1.5      PHP Content Repository interfaces
phpcr/phpcr-migrations               1.2.0      Migrations for PHPCR
phpcr/phpcr-shell                    1.1.1      Shell for PHPCR
phpcr/phpcr-utils                    1.4.1      PHP Content Repository implementation independant utilities
piwik/device-detector                3.12.5     The Universal Device Detection library, that parses User Agents and detects devices (desktop, tablet, mobile, tv, cars, console, etc.), clients (browsers...
psr/cache                            1.0.1      Common interface for caching libraries
psr/container                        1.0.0      Common Container Interface (PHP FIG PSR-11)
psr/http-client                      1.0.0      Common interface for HTTP clients
psr/http-factory                     1.0.1      Common interfaces for PSR-7 HTTP message factories
psr/http-message                     1.0.1      Common interface for HTTP messages
psr/log                              1.1.3      Common interface for logging libraries
ralouphie/getallheaders              3.0.3      A polyfill for getallheaders.
ramsey/uuid                          3.9.3      Formerly rhumsaa/uuid. A PHP 5.4+ library for generating RFC 4122 version 1, 3, 4, and 5 universally unique identifiers (UUID).
react/promise                        v2.7.1     A lightweight implementation of CommonJS Promises/A for PHP
rize/uri-template                    0.3.2      PHP URI Template (RFC 6570) supports both expansion & extraction
sentry/sdk                           2.1.0      This is a metapackage shipping sentry/sentry with a recommended http client.
sentry/sentry                        2.3.2      A PHP SDK for Sentry (http://sentry.io)
sentry/sentry-symfony                3.5.1      Symfony integration for Sentry (http://getsentry.com)
snc/redis-bundle                     3.2.2      A Redis bundle for Symfony
stof/doctrine-extensions-bundle      v1.4.0     Integration of the gedmo/doctrine-extensions with Symfony2
sulu/article-bundle                  2.0.0      Bundle for managing localized content-rich entities like blog-posts in the Sulu content management system
sulu/sulu                            2.1.0-RC1  Core framework that implements the functionality of the Sulu content management system
superbalist/flysystem-google-storage 7.2.2      Flysystem adapter for Google Cloud Storage
swiftmailer/swiftmailer              v6.2.3     Swiftmailer, free feature-rich PHP mailer
symfony-cmf/routing                  2.3.1      Extends the Symfony routing component for dynamic routes and chaining several routers
symfony-cmf/routing-bundle           2.4.0      Symfony RoutingBundle
symfony-cmf/slugifier-api            2.0.0      Provides a basic slugifier interface to transform text into strings suitable for URLs
symfony/asset                        v4.4.8     Symfony Asset Component
symfony/cache                        v4.4.8     Symfony Cache component with PSR-6, PSR-16, and tags
symfony/cache-contracts              v2.0.1     Generic abstractions related to caching
symfony/config                       v4.4.8     Symfony Config Component
symfony/console                      v4.4.8     Symfony Console Component
symfony/css-selector                 v4.4.8     Symfony CssSelector Component
symfony/debug                        v4.4.8     Symfony Debug Component
symfony/debug-bundle                 v4.4.8     Symfony DebugBundle
symfony/dependency-injection         v4.4.8     Symfony DependencyInjection Component
symfony/doctrine-bridge              v4.4.8     Symfony Doctrine Bridge
symfony/dom-crawler                  v4.4.8     Symfony DomCrawler Component
symfony/dotenv                       v4.4.8     Registers environment variables from a .env file
symfony/error-handler                v4.4.8     Symfony ErrorHandler Component
symfony/event-dispatcher             v4.4.8     Symfony EventDispatcher Component
symfony/event-dispatcher-contracts   v1.1.7     Generic abstractions related to dispatching event
symfony/expression-language          v4.4.8     Symfony ExpressionLanguage Component
symfony/filesystem                   v4.4.8     Symfony Filesystem Component
symfony/finder                       v4.4.8     Symfony Finder Component
symfony/flex                         v1.6.3     Composer plugin for Symfony
symfony/form                         v4.4.8     Symfony Form Component
symfony/framework-bundle             v4.4.8     Symfony FrameworkBundle
symfony/http-client                  v4.4.8     Symfony HttpClient component
symfony/http-client-contracts        v2.0.1     Generic abstractions related to HTTP clients
symfony/http-foundation              v4.4.8     Symfony HttpFoundation Component
symfony/http-kernel                  v4.4.8     Symfony HttpKernel Component
symfony/inflector                    v5.0.8     Symfony Inflector Component
symfony/intl                         v4.4.8     A PHP replacement layer for the C intl extension that includes additional data from the ICU library.
symfony/lock                         v5.0.8     Symfony Lock Component
symfony/mime                         v5.0.8     A library to manipulate MIME messages
symfony/monolog-bridge               v4.4.8     Symfony Monolog Bridge
symfony/monolog-bundle               v3.5.0     Symfony MonologBundle
symfony/options-resolver             v4.4.8     Symfony OptionsResolver Component
symfony/phpunit-bridge               v4.4.8     Symfony PHPUnit Bridge
symfony/polyfill-intl-grapheme       v1.16.0    Symfony polyfill for intl's grapheme_* functions
symfony/polyfill-intl-icu            v1.16.0    Symfony polyfill for intl's ICU-related data and classes
symfony/polyfill-intl-idn            v1.16.0    Symfony polyfill for intl's idn_to_ascii and idn_to_utf8 functions
symfony/polyfill-intl-normalizer     v1.16.0    Symfony polyfill for intl's Normalizer class and related functions
symfony/polyfill-mbstring            v1.16.0    Symfony polyfill for the Mbstring extension
symfony/polyfill-php73               v1.16.0    Symfony polyfill backporting some PHP 7.3+ features to lower PHP versions
symfony/polyfill-uuid                v1.16.0    Symfony polyfill for uuid functions
symfony/process                      v4.4.8     Symfony Process Component
symfony/property-access              v4.4.8     Symfony PropertyAccess Component
symfony/proxy-manager-bridge         v4.4.8     Symfony ProxyManager Bridge
symfony/routing                      v4.4.8     Symfony Routing Component
symfony/security                     v4.4.8     Symfony Security Component
symfony/security-bundle              v4.4.8     Symfony SecurityBundle
symfony/serializer                   v4.4.8     Symfony Serializer Component
symfony/service-contracts            v2.0.1     Generic abstractions related to writing services
symfony/stopwatch                    v4.4.8     Symfony Stopwatch Component
symfony/string                       v5.0.8     Symfony String component
symfony/swiftmailer-bundle           v3.4.0     Symfony SwiftmailerBundle
symfony/templating                   v4.4.8     Symfony Templating Component
symfony/thanks                       v1.2.5     Encourages sending ⭐ and 💵 to fellow PHP package maintainers (not limited to Symfony components)!
symfony/translation                  v4.4.8     Symfony Translation Component
symfony/translation-contracts        v2.0.1     Generic abstractions related to translation
symfony/twig-bridge                  v4.4.8     Symfony Twig Bridge
symfony/twig-bundle                  v4.4.8     Symfony TwigBundle
symfony/validator                    v4.4.8     Symfony Validator Component
symfony/var-dumper                   v5.0.8     Symfony mechanism for exploring and dumping PHP variables
symfony/var-exporter                 v5.0.8     A blend of var_export() + serialize() to turn any serializable data structure to plain PHP code
symfony/web-profiler-bundle          v4.4.8     Symfony WebProfilerBundle
symfony/web-server-bundle            v4.4.8     Symfony WebServerBundle
symfony/yaml                         v4.4.8     Symfony Yaml Component
toflar/psr6-symfony-http-cache-store 2.3.1      An alternative store implementation for Symfony's HttpCache reverse proxy that supports auto-pruning of expired entries and cache invalidation by tags.
twig/twig                            v2.12.5    Twig, the flexible, fast, and secure template language for PHP
willdurand/jsonp-callback-validator  v1.1.0     JSONP callback validator.
willdurand/negotiation               v2.3.1     Content Negotiation tools for PHP provided as a standalone library.
zendframework/zend-code              3.4.1      Extensions to the PHP Reflection API, static code scanning, and code generation
zendframework/zend-eventmanager      3.2.1      Trigger and listen to events within a PHP application
```

**Comment by mareksuscak** [comment]
I wonder if the route naming convention has changed at some point - FOSRestBundle's documentation kind of confirms this is expected 👇

![image](https://user-images.githubusercontent.com/613647/81538017-65185300-936e-11ea-908b-e7aa1f24b853.png)

ref: https://symfony.com/doc/current/bundles/FOSRestBundle/5-automatic-route-generation_single-restful-controller.html#implicit-resource-name-definition

**Comment by alexander-schranz** [comment]
@mareksuscak Thank you, the problem seems the `doctrine/inflector`: `1.4.0` release. The issue was fixed there https://github.com/doctrine/inflector/pull/142 but is not released yet. You can fix it by downgrading to 1.3.1 currently.

**Comment by mareksuscak** [comment]
Good catch!

**Comment by danrot** [comment]
Fixed in #5284. Going to release that today for all versions.
